Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp

```diff
@@ -271,6 +274,93 @@ void ApproachAngleWithAccel(RotationalState* state, const Common::Vec3& angle_ta
   }
 }
 
+static Common::Vec3 NormalizeAngle(Common::Vec3 angle)
+{
+  // TODO: There must be a more elegant way to do this
+  angle.x = fmod(angle.x, float(MathUtil::TAU));
+  angle.y = fmod(angle.y, float(MathUtil::TAU));
+  angle.z = fmod(angle.z, float(MathUtil::TAU));
+  angle.x += angle.x < 0 ? float(MathUtil::TAU) : 0;
+  angle.y += angle.y < 0 ? float(MathUtil::TAU) : 0;
+  angle.z += angle.z < 0 ? float(MathUtil::TAU) : 0;
+  return angle;
+}
+
+static Common::Vec3 ComplementaryFilter(const Common::Vec3& angle,
+                                        const Common::Vec3& accelerometer, const Common::Vec3& gyro,
+                                        float time_elapsed)
+{
+  Common::Vec3 gyroangle = angle + gyro * time_elapsed;
+  gyroangle = NormalizeAngle(gyroangle);
+
+  // Calculate accelerometer tilt angles
+  Common::Vec3 accangle = gyroangle;
+  if (accelerometer.y != 0 || accelerometer.z != 0)
+  {
+    if (accelerometer.x != 0 || accelerometer.z != 0)
```

Comment:
This nested `if` looks strange, since half of its condition has already been checked; and there is no code outside it.

Can you combine them into
```c++
if (accelerometer.x != 0 || accelerometer.y != 0 || accelerometer.z != 0)
{
  // code omitted
}
```

...or is there a reason to leave it like that? If so, please document it with a comment.

Reply:
I am actually unsure if the suggested code is logically equivant. Is there anyone with a better head than me who can confirm or refute?

Reply:
No, it's not logically equivalent.

Reply:
¯\\\_(ツ)\_/¯

My bad, `if ((accelerometer.x != 0 && accelerometer.y != 0) || accelerometer.z != 0)` then.